### ISSUE TITLE
Build problems due to $GOPATH reversal

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#335]
- in 261b30bb we moved to using `go clean -r`, which recursively also cleans dependencies. Unfortunately, that includes the standard library, which resides in `/usr/local/go` (at least when using `godeb`) and isn't accessible by user accounts. Hence invocations of `make clean` will fail.

  It's the same problem [this guy](http://stackoverflow.com/questions/23884525/go-clean-i-r-tries-to-remove-usr-local-go-pkg-linux-amd64-runtime-a) is having. For `go build -a`, there was a related change here: [make build -a skip standard packages in Go releases](https://code.google.com/p/go/source/detail?r=6364b071649c).

  The workaround for now is to use `sudo GOPATH=$GOPATH make clean`, but then I also need to run `make build` via `sudo` because I won't have a standard library any more.

- in 37dbbcb9 we switched the order of the vendored and global $GOPATH variables. That means I'm not allowed to have coreos/etcd in my global GOPATH or it will prefer that over the vendored version.

**Comment by bdarnell** [comment]
I don't think the `-r` to `go clean` is actually helping us, since it appears to just delete executables that appear in the source tree, not stuff in `$GOPATH/pkg`.

Good point about the GOPATH switch; I no longer use a global GOPATH so I hadn't noticed the mistake when I made that change. The reason I did it is that bootstrap.sh is run with the environment's GOPATH, which meant that gogoproto was getting installed into `_vendor` for me but `../../../..` for a more "typical" configuration. We should probably set GOPATH in bootstrap.sh to aways use _vendor when installing dependencies. (I had actually started a branch that moved all dependencies into submodules under _vendor, but ran into troubles with the Dockerfile).
